.github/actions/upload_terraform_module/action.yml

```diff
@@ -0,0 +1,23 @@
+name: Upload Terraform module
+description: "Upload the Terraform module as an artifact."
+runs:
+  using: "composite"
+  steps:
+    - name: Copy Terraform module
+      shell: bash
+      run: |
+        cp -r terraform terraform-module
+        rm terraform-module/assets.go terraform-module/BUILD.bazel
```

Comment:
I added rules to delete all *.go and all *.bazel. Deleting all but .tf is too dangerous, since there are different file formats used: lock files, bash scripts, xsl file and who knows whatelse in the future.